setup.py

```diff
@@ -58,6 +59,10 @@ def _minimal_ext_cmd(cmd):
 
 
 def write_version_py(filename='traits/_version.py'):
+    """
```

Comment:
Possible nit-picky comment - the function is `write_version_py` but now it's being used to get the full version info, with the side-effect of also writing the version file.

It might be cleaner and more readable to split this up into a `get_version`, which uses `write_version_py` in the absence of a `traits/_version.py` file.

Reply:
Yes, it's a mess that needs to be rewritten. I didn't really want to do that in this PR - this was supposed to be a quick fix for an actual bug. I'll take a look.

Reply:
I've just been trying to do a minimal refactor, but there's no such thing: as soon as I try to refactor, things get messy. I'm going to leave this as it is for this PR and delegate reworking the version machinery to a separate PR.